### PR TITLE
[Feat] Add 4o-mini to model list

### PR DESCRIPTION
**Pull request by lingyielia**
## Description
Did some quick experiments on dashboard tasks. 4o-mini can create simple dashboard. When requirements become complex, the dashboard generated doesn't look nice. 

Already noticed that 4o-mini performed consistently worse in 2 sub-tasks:
1. understand the "grid-template-areas" for layout plan
2. follow the instruction to give datasets unique names

It still worth adding it to the list though, because it's cheap and can perform simple dashboard tasks.

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ Vizro-AI currently supports the following [OpenAI models](https://platform.opena
     - gpt-3.5-turbo-1106
     - gpt-3.5-turbo-0125
     - gpt-4o-2024-05-13
+    - gpt-4o-mini-2024-07-18
+    - gpt-4o-mini
```
Comment (maxschulz-COL) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:23]: Can we also double check the other models, if they are still up-to-date?

**Review by maxschulz-COL — APPROVED** [review]

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md [hunk 2]
```diff
@@ -33,9 +35,9 @@ These models offer different levels of performance and cost to Vizro-AI users.
 
 ### Chart generation
 
-For chart creation with OpenAI, the **gpt-3.5** model series have lower price points and faster speeds for providing answers, but do not offer sophisticated charting.
+For chart creation with OpenAI, the **gpt-4o-mini** and **gpt-3.5** model series have lower price points and faster speeds for providing answers, but do not offer sophisticated charting.
```
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:38]:
```suggestion
For basic chart creation with OpenAI, the **gpt-4o-mini** and **gpt-3.5** model series have lower price points and faster speeds, but they do not offer sophisticated charting.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md [hunk 3]
```diff
@@ -33,9 +35,9 @@ These models offer different levels of performance and cost to Vizro-AI users.
...
 
-Consider upgrading to the **gpt-4** and **gpt-4o** model series for more demanding tasks. While they are part of a more capable GPT model series, they come at a higher cost.
+Consider upgrading to the **gpt-4o** and **gpt-4** model series for more demanding tasks. While they are part of a more capable GPT model series, they come at a higher cost.
```
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:40]:
```suggestion
Consider upgrading to the **gpt-4o** and **gpt-4** model series for more demanding tasks. The downside of using these models is that they come at a higher cost.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md [hunk 4]
```diff
@@ -46,8 +48,9 @@ Refer to the [OpenAI documentation for more about model capabilities](https://pl
 === "OpenAI"
 
     - gpt-3.5-turbo `default`: Ideal for smaller requests and simple specifications. However, due to the complexity of dashboard creation, `gpt-3.5-turbo` often produces incomplete responses for larger tasks.
-    - gpt-4-turbo: Excels in dashboard creation tasks, providing the most stable and complete responses.
-    - gpt-4o: Similar performance in terms of the quality of generated content. Compared to gpt-4-turbo, gpt-4o is much faster and cheaper.
+    - gpt-4o: Most ideal for dashboard generation tasks. Excels in dashboard creation tasks, providing stable and complete responses. Compared to gpt-4-turbo, gpt-4o is much faster and cheaper.
```
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:51]:
```suggestion
    - gpt-4o: Most ideal for dashboard generation tasks, providing stable and complete responses. Compared to gpt-4-turbo, gpt-4o it is much faster and cheaper.
```

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md [hunk 5]
```diff
@@ -46,8 +48,9 @@ Refer to the [OpenAI documentation for more about model capabilities](https://pl
...
+    - gpt-4-turbo: Excels in dashboard creation tasks, providing stable and complete responses.
+    - gpt-4o-mini: Ideal for smaller requests and simple specifications. Most cost-effective. However, due to the complexity of dashboard creation, its performance is not stable.
```
Comment (stichbury) [inline vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:53]:
```suggestion
    - gpt-4o-mini: Ideal for smaller requests and simple specifications. Most cost-effective. However, due to the complexity of dashboard creation, its performance is not guaranteed to be reliable.
```

Comment (stichbury) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:53]: I went with "reliable" rather than "stable" is that OK?

**Review by stichbury — APPROVED** [review]
LGTM 🌟  Just a few minor suggested modifications.

File: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md (hunk 1 above)
Comment (lingyielia) [reply vizro-ai/docs/pages/user-guides/customize-vizro-ai.md:23]: Just double checked, everything is up to date so far 👍

**Pull request merged**